lib/web3/function.js

```diff
@@ -37,7 +37,7 @@ var SolidityFunction = function (eth, json, address) {
     this._outputTypes = json.outputs.map(function (i) {
         return i.type;
     });
-    this._constant = json.constant;
+    this._constant = (json.stateMutability === "view" || json.stateMutability === "pure" || json.constant);
     this._payable = json.payable;
```

Comment:
Also this should check for `stateMutability === "payable"` 😉

Reply:
Well, yes :). We haven't removed ``payable`` yet, but yes, that would be the next step - I'll include it in this PR already.